pkg/api/types.go

```diff
@@ -257,6 +259,8 @@ type PersistentVolumeSource struct {
 	Flocker *FlockerVolumeSource `json:"flocker,omitempty"`
 	// AzureFile represents an Azure File Service mount on the host and bind mount to the pod.
 	AzureFile *AzureFileVolumeSource `json:"azureFile,omitempty"`
+	// DockerVolume represents a volume that is backed by a Docker volume plugin. This depends on the plugin being already properly setup.
+	DockerVolume *DockerVolumeSource `json:"dockerVolume,omitempty"`
```

Comment:
I would take this out of PVSource - we don't have enough information to do this, yet.